src/app/app.component.ts

```diff
@@ -11,12 +12,29 @@ import { HomePage } from '../pages/home/home';
 export class MyApp {
   rootPage:any = HomePage;
 
-  constructor(platform: Platform, statusBar: StatusBar, splashScreen: SplashScreen) {
+  @ViewChild(Nav) navChild:Nav;
+
+  constructor(
+    private platform: Platform, 
+    private deeplinks : Deeplinks,
+    statusBar: StatusBar, 
+    splashScreen: SplashScreen) {
     platform.ready().then(() => {
       // Okay, so the platform is ready and our plugins are available.
       // Here you can do any higher level native things you might need.
       statusBar.styleDefault();
-      splashScreen.hide();
+      splashScreen.hide();    
     });
   }
+
+  ngAfterViewInit() {
+    this.deeplinks.routeWithNavController(this.navChild, {
+      '/': HomePage
+    }).subscribe((match) => {
+      console.log('Successfully routed\n'+ JSON.stringify(match));
+    }, (nomatch) => {
```

Comment:
**Best Practice:** For an anonymous function, when there's only 1 parameter, it is optional to add the parenthesis around them. In my opinion, it looks much cleaner without the parenthesis.

Reply:
Thanks! Slipped right past me.